core-tests/src/test/java/org/mule/runtime/core/registry/RegisteredObjectLifecycleTestCase.java

```diff
@@ -46,17 +46,20 @@ public void testLifecycleForMuleContext() throws Exception
     {
         muleContext.getRegistry().registerObject("dummy", bean);
         assertTrue(initLatch.await(TIMEOUT, TimeUnit.MILLISECONDS));
+        Thread.sleep(TIMEOUT);
```

Comment:
Actually, this is testing the state of the conditions after a certain time. With a prober, the first time it runs they would pass the assertion and we wouldn't be testing what happens after a certain time